src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -264,6 +264,21 @@ Vfs::AvailabilityResult VfsCfApi::availability(const QString &folderPath)
     return AvailabilityError::NoSuchItem;
 }
 
+void VfsCfApi::cancelHydration(const QString &requestId, const QString & /*path*/)
+{
+    // Find matching hydration job for request id
+    const auto hydrationJobsIter = std::find_if(d->hydrationJobs.cbegin(), d->hydrationJobs.cend(), [&](const HydrationJob *job) {
+        return job->requestId() == requestId;
+    });
+
+    if (hydrationJobsIter == d->hydrationJobs.cend()) {
```

Comment:
@FlexW I don't think this is necessary, as there won't be a lot of additional logic to make it reasonable to early-out.

You can simply write:
```
if (!hydrationJobsIter != d->hydrationJobs.cend()) {
  (*hydrationJobsIter)->cancel();
}
```

Optionally, since it's still going to be a linear search, you can ditch the `std::find_if` and reduce the amount of code even further by just writing:

```
for (const auto it = std::cbegin(d->hydrationJobs);  it != std::cend(d->hydrationJobs); ++it) {
    if (job->requestId() == requested) {
        (*hydrationJobsIter)->cancel();
        break;
    }
}
```
Or, just use `for(const auto &hydrationJob : qAsConst(hydrationJobs))` form. Whatever :)

Reply:
I thought I use `std::find_if` instead of a loop because it's more expressive. If you see `std::find_if` you know directly whats going on. With the loop you have to look closer. Of course that depends a bit on my personal taste. Also I think the main task of the function is to cancel the hydration. The if statement is there to catch a case that should not happen if the function is used correctly. I don't like to put the main logic/exercise of a function in a if branch.

Reply:
> Also I think the main task of the function is to cancel the hydration. The if statement is there to catch a case that should not happen if the function is used correctly. I don't like to put the main logic/exercise of a function in a if branch :)

@FlexW Well, I can agree with most of your thoughts, but, the one with not moving the main logic into the branch, I don't think it applies in the current case.

The thing is, you are adding a redundant line with a condition here:
```
if (hydrationJobsIter == d->hydrationJobs.cend()) {
    return;
}
```
It increases the amount of code and does exactly what the end of a function would do - it returns from the function. I would agree to your comment, in case the function would return a value. But, it is just a void in our case, and, adding an additional return statement does not help the readability. Also, the branch is not that deep (it's just one if).